### ISSUE TITLE
Suspicious virtual address arithmetic

### ISSUE THREAD

**Issue by ladislav-zezula** [avast/pelib#1]
In the `DelayImportDirectory::read()` function (DelayImportDirectory.h), there is a suspicious arithmetic which possibly subtracts a 64-bit image base from a 32-bit value:
```
	if (std::abs(static_cast<int>(uiDelayImportsVa - rec.NameRva)) <
	    std::abs(static_cast<int>(uiDelayImportsRva - rec.NameRva)))
	{
		rec.NameRva -= peHeader.getImageBase();
	}
```
The meaning of the entire block is a mystery to me (perhaps parsing files that are memory image dumps?). However, it may lead to undefined results, as image bases in 64-bit PE files are also 64-bit and typically exceed the 4 GB boundary (default image base for files produces by MSVC linker is 00000001-40000000).

**Comment by metthal** [comment]
The mentioned code probably tries to solve [this](https://reverseengineering.stackexchange.com/questions/16261/should-the-delay-import-directory-contain-virtual-addresses) in the very hacky way. We should use the attribute mentioned in the SO answer to recognize whether delayed import directory contains VAs or RVAs.

**Comment by s3rvac** [comment]
Possibly related to https://github.com/avast-tl/retdec/issues/282.

**Comment by ladislav-zezula** [comment]
Fixed in commit ba7126dc13bd378c5360b68b0141488d5ce18d52
